examples/todo/test/acceptance/application.acceptance.ts

```diff
@@ -19,6 +19,20 @@ describe('Application', () => {
   before(givenAnApplication);
   before(async () => {
     await app.boot();
+
+    /**
+     * Override DataSource to not write to file for testing. Since we aren't
+     * persisting data to file and each injection normally instatiates a new
+     * instance, we must change the BindingScope to a singleton so only one
+     * instance is created and used for all injections (preserving access to
+     * the same memory space).
+     */
+    app.bind('datasources.config.db').to({
+      name: 'db',
+      connector: 'memory',
+    });
```

Comment:
`.inScope(BindingScope.SINGLETON)` here? (IIUC from the comment above)

Reply:
Only a `DataSource` instance needs to be bound as a singleton. The config -- which is just the declarative JSON config can be bound as normal to the context.